driver-core/src/test/functional/com/mongodb/client/model/expressions/ArithmeticExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,210 @@
+
+class ArithmeticExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
+    // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#arithmetic-expression-operators
+
+    @Test
+    public void literalsTest() {
+        assertExpression(1, of(1), "1");
+        assertExpression(1L, of(1L));
+        assertExpression(1.0, of(1.0));
+
+        // expression equality differs from bson equality
+        assertExpression(true, of(1L).eq(of(1.0)));
+        assertExpression(true, of(1L).eq(of(1)));
+
+        // bson equality; underlying type is preserved
+        // this is not defined by the API, but tested for clarity
+        assertEquals(toBsonValue(1), evaluate(of(1)));
+        assertEquals(toBsonValue(1L), evaluate(of(1L)));
+        assertEquals(toBsonValue(1.0), evaluate(of(1.0)));
+        assertNotEquals(toBsonValue(1), evaluate(of(1L)));
+        assertNotEquals(toBsonValue(1.0), evaluate(of(1L)));
+    }
+
+    @Test
+    public void multiplyTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/multiply/
+        assertExpression(
+                2.0 * 2,
+                of(2.0).multiply(of(2)),
+                "{'$multiply': [2.0, 2]}");
+
+        // mixing integers and numbers
+        IntegerExpression oneInt = of(1);
+        NumberExpression oneNum = of(1.0);
+        IntegerExpression resultInt = oneInt.multiply(oneInt);
+        NumberExpression resultNum = oneNum.multiply(oneNum);
+        // compile time error if these were IntegerExpressions:
+        NumberExpression r2 = oneNum.multiply(oneInt);
+        NumberExpression r3 = oneInt.multiply(oneNum);
+        assertExpression(1, resultInt);
+        // 1 is also a valid expected value in our API
+        assertExpression(1.0, resultNum);
+        assertExpression(1.0, r2);
+        assertExpression(1.0, r3);
+
+        // convenience
+        assertExpression(2.0, of(1.0).multiply(2.0));
+        assertExpression(2, of(1).multiply(2));
+    }
+
+    @Test
+    public void divideTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/divide/
+        assertExpression(
+                1.0 / 2.0,
+                of(1.0).divide(of(2.0)),
+                "{'$divide': [1.0, 2.0]}");
+        // unlike Java's 1/2==0, dividing any type of numbers always yields an
+        // equal result, in this case represented using a double.
+        assertExpression(
+                0.5,
+                of(1).divide(of(2)),
+                "{'$divide': [1, 2]}");
+        // divide always returns a Number, so the method is not on IntegerExpression
+
+        // convenience
+        assertExpression(0.5, of(1.0).divide(2.0));
+        assertExpression(0.5, of(1).divide(2.0));
+        // TODO these convenience methods are all for doubles, so while they
+        // TODO allow longs and ints, they do not carry that through to the server:
+        assertExpression(0.5, of(1).divide(2L));
```

Comment:
I have created convenience methods only for doubles. If it is critical that the underlying structure, a long, is correctly propagated into the server, the user would need to use `of`.